R/simmicmenten.R

```diff
@@ -0,0 +1,42 @@
+simmicmenten <- function(x, vmax, km, cv, reps, h = 1, log=F) {
+
+  y = vmax*x^h/(km^h + x)
+
+  values <- data.frame(x=rep(x, reps), y)
+
+  values <- dplyr::mutate(values, y=apply(values, 1, function(x) stats::rnorm(1, x[2], cv*x[2])))
+
+  for (i in 1) {
+    if (log) {
+      model <- stats::lm(log(y) ~ log(x), data=values)
+      start <- list(vmax=exp(stats::coef(model)[1]), km=exp(stats::coef(model))[2])
+      model <- stats::nls(y ~ vmax * x^h/(km^h + x), data = values, start = start)
+      vmax.var <- model$m$getPars()[1]
+      km.var <- model$m$getPars()[2]
+      break
+    } else {break}
+  }
+
+  ggplot2::ggplot(
+    values,
+    ggplot2::aes(x=if (log){
+      log10(x)} else {
+        x}, y)) +
+    ggplot2::geom_point(size=2) +
+    ggplot2::labs(title="model: y=vmax*x^h/(x+km^h)") +
+    if (log) {
+      ggplot2::stat_function(geom = "smooth", fun = function(x) y = vmax.var*10^x^h/(km.var^h + 10^x),
```

Comment:
We run into the same issue as with sim1sbind. Maybe it is better to have a function dedicated to log, and a separate function dedicated to linear? 

There is this old saying from Jared, "write functions that do one thing well. try not to make them do two things"

Reply:
I really do think that this will work for both. I left a more detailed comment about this over on the sim1sbind pull request thread. 

I understand what you mean by having a function do two things, but I think that choosing between a log-transformed x vector or a linear x vector doesn't seem like this function (or sim1sbind) is trying to do too much